### PR TITLE
Verify metapack for consistency before saving

### PR DESCRIPTION
**Pull request by patriotyk**

### CODE REVIEW

**Comment by OYTIS** [comment]
To me it seems contrary to the direction of refactoring suggested by Eric, namely isolating storage and network communication from TUF/Uptane-related logic (see for example, https://advancedtelematic.atlassian.net/browse/PRO-4771).

**Comment by OYTIS** [comment]
I mean the ticket itself is OK, since we don't yet have storage for non-verified metadata. But making the check inside the class abstracting the storage just feels wrong.

**Comment by patriotyk** [comment]
But ticket requires MetaPack::IsConsistent() to be created. So should I address Your comment? Or do we need to changes the ticket?

**Comment by OYTIS** [comment]
Having isConsistent() as a member of MetaPack is probably OK for now. But I strongly believe that the check (`if(pack.isConsistent()) { ...`) should be made by the caller of `storeMetadata`, not from inside the storage classes.

**Comment by cajun-rat** [comment]
@OYTIS Are you happy with this PR now?

File: src/libaktualizr/uptane/managedsecondary.cc [hunk 1]
```diff
@@ -27,6 +27,9 @@ bool ManagedSecondary::putMetadata(const MetaPack &meta_pack) {
   // No verification is currently performed, we can add verification in future for testing purposes
   detected_attack = "";
   current_meta = meta_pack;
+  if (current_meta.isConsistent()) {
```
Comment (lbonn) [inline src/libaktualizr/uptane/managedsecondary.cc]: should be `if (!current_meta.isConsistent())`?

Comment (patriotyk) [reply src/libaktualizr/uptane/managedsecondary.cc]: Yes, thank You.

**Comment by OYTIS** [comment]
Looks like the check is only done for secondaries. Is it intended?

**Comment by cajun-rat** [comment]
Rebased

**Comment by patriotyk** [comment]
@OYTIS It is because we already do more advanced checks for the primaries.

**Pull request merged**